### PR TITLE
Add cluster validation interceptor to resolve the Cross-Cluster routing issue (#25001)

### PR DESCRIPTION
**Pull request by bigsheeper**
issue: https://github.com/milvus-io/milvus/issues/25000

/kind bug

### CODE REVIEW

File: internal/util/grpcclient/client.go [hunk 1]
```diff
@@ -273,6 +288,10 @@ func (c *ClientBase[T]) callOnce(ctx context.Context, caller func(client T) (any
 		go c.bgHealthCheck(client)
 		return generic.Zero[T](), err
 	}
+	if IsCrossClusterRoutingErr(err) {
+		c.resetConnection(client)
+		return ret, interceptor.ErrServiceUnavailable // For concealing ErrCrossClusterRouting from the client
```
Comment (czs007) [inline internal/util/grpcclient/client.go:293]: add log.Info/log.Warn about this error.

**Comment by bigsheeper** [comment]
rerun ut

**Comment by bigsheeper** [comment]
rerun ut

**Pull request merged**